### ISSUE TITLE
How did you deal with Squirrel.Windows update loop?

### ISSUE THREAD

**Issue by mzmousa** [Aluxian/Whatsie#242]
Question: Did you have to deal with the Squirrel.Windows autoUpdater downloading files twice when doing updates? If yes, how did you deal with it, or did you find a way around it?

**Comment by Aluxian** [comment]
Hi! Is that a known bug in Electron or Squirrel.Windows? I'm not aware of apps (or Whatsie) downloading Windows updates twice.

**Comment by mzmousa** [comment]
Yes! There's an issue open that I'll reference here (soon) on Squirrel.Windows, but everytime I do an autioUpdate on my app using Windows, it downloads the latest nupkg to Packages twice before updating. What are you using to generate your windows artifacts? I'm using electron builder

**Comment by Aluxian** [comment]
I use my own suite of gulp tasks, together with my [electron-windows-installer](https://github.com/Aluxian/electron-windows-installer) module. Builds run on AppVeyor, so everything is automated.

**Comment by mzmousa** [comment]
I see, thanks!

**Comment by Aluxian** [comment]
What was the issue you wanted to reference?

**Comment by mzmousa** [comment]
Sorry, forgot to do that. It's referenced a few times [here](https://github.com/electron/electron/issues/5057), [here](https://github.com/electron-userland/electron-builder/issues/359), and [here](https://github.com/electron-userland/electron-builder/issues/234). I get this error myself, but it's not quite an issue if you just wait it out...it just causes problems with some of autoUpdater's event listeners for me

**Comment by Aluxian** [comment]
Do you have both 32 and 64 bit builds?

**Comment by mzmousa** [comment]
No, only 64 bit.

**Comment by Aluxian** [comment]
Maybe that's why... I have only 32-bit builds.

**Comment by mzmousa** [comment]
I see! If you're not coming across that issue then this might be worth closing. I'll definitely try seeing if I get this on 32-bit builds.
Big fan of your work BTW. Great job on Whatsie so far :grin:

**Comment by develar** [comment]
@mzmousa Don't worry, electron-builder NSIS target will support auto-update soon and it will be smaller and faster than Squirrell.Windows.

**Comment by Aluxian** [comment]
@mzmousa The only issue I have with Squirrel.Windows is that when there's an update available it downloads it until the user accepts to install. But if the user quits without updating, I don't know where that file goes, because the app has to download it again... But I think I know how to fix this (auto-update on exit).

How did you test if your apps download the update twice? I want to test Whatsie too, but I don't have any tools installed for that...

Thanks! 😊

**Comment by mzmousa** [comment]
Sorry, I meant to reply to this earlier.
I'm not using any tools to test if the app downloads the update twice - I just see it happen myself in `AppData/Local/MyApp/Packages`. The latest `-full.nupkg` or `-delta.nupkg` is downloaded to full, then something happens, then it goes to 0, then it downloads to full again, and only _then_ do I see my app starting to apply the update.

**Comment by Aluxian** [comment]
Hey @mzmousa, I've just checked and I also have double downloads:

![image](https://cloud.githubusercontent.com/assets/1418870/16927592/c70f4c9e-4d36-11e6-8ce6-5adfbfe84a1b.png)

As you can see, there were 4 requests...

**Comment by mzmousa** [comment]
Thanks for checking it out. It's been a known issue on Squirrel.Windows for quite a while now, but it never seemed to be much of a problem, so it was never really addressed. I'll keep watch and update you if I come across anything :)

**Comment by Aluxian** [comment]
Upstream: https://github.com/electron/electron/issues/5057
